service/resource/cloudformation/main_stack.go

```diff
@@ -2,19 +2,27 @@ package cloudformation
 
 import (
 	"bytes"
+	"strconv"
 	"text/template"
 
-	awscloudformation "github.com/aws/aws-sdk-go/service/cloudformation"
 	"github.com/giantswarm/aws-operator/service/key"
 	"github.com/giantswarm/awstpr"
 )
 
 func newMainStack(customObject awstpr.CustomObject) (StackState, error) {
 	stackName := key.MainStackName(customObject)
+	workers := len(customObject.Spec.AWS.Workers)
+	var imageID string
+	if workers > 0 {
+		imageID = customObject.Spec.AWS.Workers[0].ImageID
```

Comment:
Let's create an issue for restructuring awstpr and have a comment here.

Reply:
Sure, done